src/patterns/structural/small-crates.md

```diff
@@ -26,10 +26,35 @@ fine-grained dependencies.
   has both versions 1.0 and 0.5. Since the `Url` from `url:1.0` and the `Url`
   from `url:0.5` are different types, an HTTP client that uses `url:0.5` would
   not accept `Url` values from a web scraper that uses `url:1.0`.
-- Packages on crates.io are not curated. A crate may be poorly written, have
+  Rust and Cargo provide several options to handle this disadvantage:
+  - [workspace.dependencies](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-dependencies-table)
+    helps in enforcing the same crate version across all workspace members
+  - dependnecy can be renamed in cargo.toml and in Rust module import 
+  - `package.resolver= "2"` allows to have several indirect and direct dependncies to be in same workspace,
+     so sometimes it fails (for few reasons), one can:
+    - if resolver fails to find compatible versions,
+      can use [patch](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section)
+      to pin dependecy accross `Cargo.lock`
+    - `resolver = "3"` added Rust-version awareness,
+       so incompatible crate versions (based on the `rust-version` field) are not selected,
+       further reducing potential dependency hell
```

Comment:
isn't the latest resolver already the default of the last rust edition? If yes, why documenting this? People basically don't need to know I think because the default is already the best option, right?